tests/test_boolean.py

```diff
@@ -324,217 +195,419 @@ def test_parse_side_by_side_symbols_with_parens_raise_exception(self):
         l1 = Symbol(1)
         l2 = Symbol(1)
-        self.assertTrue(l1 in l1.literals)
-        self.assertTrue(l1 in l2.literals)
-        self.assertTrue(l2 in l1.literals)
-        self.assertTrue(l2 in l2.literals)
-        self.assertRaises(AttributeError, setattr, l1, 'literals', 1)
+
+        assert l1 in l1.literals
+        assert l1 in l2.literals
+        assert l2 in l1.literals
+        assert l2 in l2.literals
+
+        for symbol in [l1, l2]:
+            with pytest.raises(AttributeError):
+                symbol.setattr('literals', 1)
 
     def test_literalize(self):
         s = Symbol(1)
-        self.assertEqual(s.literalize(), s)
+
+        assert s.literalize() == s
 
     def test_simplify(self):
         s = Symbol(1)
-        self.assertEqual(s.simplify(), s)
 
-    def test_equal_symbols(self):
+        assert s.simplify() == s
+
+    def test_symbols_eq_0(self):
+        algebra = BooleanAlgebra()
+
+        a = algebra.Symbol('a')
+
+        assert a == a
+
+    def test_symbols_eq_1(self):
         algebra = BooleanAlgebra()
+
+        a0 = algebra.Symbol('a')
+        a1 = algebra.Symbol('a')
+
+        assert a0 == a1
+        assert a1 == a0
+
+    def test_symbols_eq_2(self):
+        algebra = BooleanAlgebra()
+
         a = algebra.Symbol('a')
-        a2 = algebra.Symbol('a')
+        b = algebra.Symbol('b')
 
-        c = algebra.Symbol('b')
-        d = algebra.Symbol('d')
-        e = algebra.Symbol('e')
+        assert not a == b
+        assert not b == a
 
-        # Test __eq__.
-        self.assertTrue(a == a)
-        self.assertTrue(a == a2)
-        self.assertFalse(a == c)
-        self.assertFalse(a2 == c)
-        self.assertTrue(d == d)
-        self.assertFalse(d == e)
-        self.assertFalse(a == d)
-        # Test __ne__.
-        self.assertFalse(a != a)
-        self.assertFalse(a != a2)
-        self.assertTrue(a != c)
-        self.assertTrue(a2 != c)
+    def test_symbols_eq_3(self):
+        algebra = BooleanAlgebra()
+
+        assert     algebra.Symbol('a') == algebra.Symbol('a')
+        assert not algebra.Symbol('a') == algebra.Symbol('b')
+
+    def test_symbols_ne_0(self):
+        algebra = BooleanAlgebra()
+
+        a = algebra.Symbol('a')
+
+        assert not a != a
+
+    def test_symbols_ne_1(self):
+        algebra = BooleanAlgebra()
+
+        a0 = algebra.Symbol('a')
+        a1 = algebra.Symbol('a')
+
+        assert not a0 != a1
+        assert not a1 != a0
+
+    def test_symbols_ne_2(self):
+        algebra = BooleanAlgebra()
+
+        a = algebra.Symbol('a')
+        b = algebra.Symbol('b')
+
+        assert a != b
+        assert b != a
+
+    def test_symbols_ne_3(self):
+        algebra = BooleanAlgebra()
+
+        assert not algebra.Symbol('a') != algebra.Symbol('a')
+        assert     algebra.Symbol('a') != algebra.Symbol('b')
+
+    def test_symbols_eq_ne(self):
+        algebra = BooleanAlgebra()
+
+        symbols0 = [
```

Comment:
I tend to prefer many small tests methods that a loop... Now  if we keep a loop, this still might need exploding in different purpose tests?   I find it a tad hard to read what the tests are really doing